test/e2e/run.sh

```diff
@@ -481,26 +481,32 @@ get-config-generation() {
     vm-command-q "kubectl get -n kube-system $resource -ojsonpath={.metadata.generation}"
 }
 
+get-hostname-for-vm() {
+    local node="${node:-$VM_HOSTNAME}"
+    echo ${node%.*}
+}
+
 get-config-node-status-generation() {
-    local resource="$1" node="${node:-$VM_HOSTNAME}"
+    local resource="$1" node="$(get-hostname-for-vm)"
+    node="${node%.*}"
```

Comment:
491 looks suspicious, as we're cutting again the suffix after the last dot, but we don't do that after line 497 yet both $node act as keys to the same map/dict/object.